lib/pg/instrumentation.rb

```diff
@@ -0,0 +1,157 @@
+require 'pg/instrumentation/version'
+require 'opentracing'
+
+module PG
+  module Instrumentation
+    class << self
+
+      attr_accessor :tracer
+      attr_accessor :tags
+
+      ::PG::Instrumentation.tags = {
+        'span.kind' => 'client',
+        'component' => 'pg',
+        'db.type' => 'pg'
+      }
+
+      def instrument(tracer: OpenTracing.global_tracer)
+        begin
+          require 'pg'
+        rescue LoadError
+          return
+        end
+
+        @tracer = tracer
+
+        patch_methods unless @instrumented
+        @instrumented = true
+      end
+
+      def patch_methods
+        ::PG::Connection.class_eval do
+
+          alias_method :initialize_original, :initialize
+          alias_method :exec_original, :exec
+          alias_method :exec_params_original, :exec_params
+          alias_method :prepare_original, :prepare
+          alias_method :exec_prepared_original, :exec_prepared
+
+          def initialize(*args)
+            return '' if args.empty?
```

Comment:
It appears you're preventing the intended execution with these early returns instead of anticipating empty arguments in tag and span creation.  Instrumentation should really never affect the end behavior of the target lib.